test/core/iomgr/resource_quota_test.cc

```diff
@@ -827,6 +918,11 @@ int main(int argc, char** argv) {
   test_negative_rq_free_pool();
   gpr_mu_destroy(&g_mu);
   gpr_cv_destroy(&g_cv);
+
+  // Resource quota thread related
+  test_thread_limit();
+  test_thread_maxquota_change();
```

Comment:
How about a test for setting the thread quota to a negative number? Just to make sure it can't be done and also because we're not using unsigned for it. Tests for allocing and freeing a negative number of threads would also be good.

Reply:
ah..negative threads don't make sense here.. I meant to add GPR_ASSERTS() in the code. Thanks for pointing this out!! 
(I added GPR_ASSERTS now)